From 923be37aa87fcc257040ff830676c0e867b00d3d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Arve=20Hj=C3=B8nnev=C3=A5g?= <arve@android.com>
Date: Mon, 26 Nov 2012 16:23:33 -0800
Subject: [PATCH] ARM: fiq_debugger: Fix to compile on 3.7
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Use for_each_irq_desc in arch/arm/common/fiq_debugger.c

Signed-off-by: Arve Hjønnevåg <arve@android.com>
---
 arch/arm/common/fiq_debugger.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/arm/common/fiq_debugger.c b/arch/arm/common/fiq_debugger.c
index 053680b..5e9005b 100644
--- a/arch/arm/common/fiq_debugger.c
+++ b/arch/arm/common/fiq_debugger.c
@@ -374,16 +374,17 @@ static void dump_allregs(struct fiq_debugger_state *state, unsigned *regs)
 static void dump_irqs(struct fiq_debugger_state *state)
 {
 	int n;
+	struct irq_desc *desc;
 
 	debug_printf(state, "irqnr       total  since-last   status  name\n");
-	for (n = 0; n < NR_IRQS; n++) {
-		struct irqaction *act = irq_desc[n].action;
+	for_each_irq_desc(n, desc) {
+		struct irqaction *act = desc->action;
 		if (!act && !kstat_irqs(n))
 			continue;
 		debug_printf(state, "%5d: %10u %11u %8x  %s\n", n,
 			kstat_irqs(n),
 			kstat_irqs(n) - state->last_irqs[n],
-			irq_desc[n].status_use_accessors,
+			desc->status_use_accessors,
 			(act && act->name) ? act->name : "???");
 		state->last_irqs[n] = kstat_irqs(n);
 	}
-- 
1.8.0

